### ISSUE TITLE
Episodes are being added before they are aired.

### ISSUE THREAD

**Issue by ElMouse** [scakemyer/plugin.video.quasar#807]
Hello. I do not know if that is an issue. If this is not an issue please kindly treat as request for enhancement.
The following description is an example. The same is true for all shows added to my library.

The Quasar settings 
"show  unaired episodes"
"show unaired seasons"
are swithed off.

It is currently 13 of March 14-00 at my place (GMT +3 timezone)
The episode of show "2 Broke Girls" episode "And the Baby and Other Things" is aired
AIRS March 13, 2017 21:30 on CBS as accriding to Trakt. (GMT -5 Timezone)

So it is currently 13 March 8am there
And the show will air on 14 March 4 am in my timezone.  In 14 hours

And the show is already added to my library - without the description, screenshot and so on.

So basically what i am saying the show is added too early. It will not air for quite some time.

It would be really cool if the shows would be added after the actual TIME of the episode airs, accounting for difference in time zones.

If it is not possible or very difficult to add this functionality, maybe you could add a setting for delay into Quasar.
Something like:
Add episodes <1> days after the air date.

Thank you!

**Comment by KillerJoeBR** [comment]
That would be nice.  Right now episodes are being added with one day before the release date and because of this they are added to kodi without screenshots description etc a lot of time. Just like Elmouse said. 

Then I need to keep updating the info later to get all the details missing.

**Comment by nix87** [comment]
I don't have problem with this. My episodes are added right in time. I am +1 GMT.

**Comment by KillerJoeBR** [comment]
Episodes are being added for me as soon as the day of the air time begins. So right now, episode 6 of legion got added to my library. Like 20 hours early or more.

**Comment by ElMouse** [comment]
Yes, same for me as for KillerJoeBR and as i described. 
I wonder what is the difference between you, nix87 and us.

**Comment by scakemyer** [comment]
Time zones is the difference ;) I wanted to fix this for a while; will try to do so for the next release.

**Comment by ElMouse** [comment]
Many thanks, scakemyer! That would be awesome. Many thanks for the Quasar!

I am kind of puzzled that KillerJoeBR at i am assuming GMT-3 has this problem,
I am at GMT+3 have this problem 
and nix87 at GMT+1 - between us does not.

**Comment by johnnyslt** [comment]
I have the same problem and I have GMT+2.

Some episodes are added even a couple of days ahead.

Pe 22 mar. 2017 12:16, "ElMouse" <notifications@github.com> a scris:

Many thanks, scakemyer! That would be awesome. Many thanks for the Quasar!

I am kind of puzzled that KillerJoeBR at i am assuming GMT-3 has this
problem,
I am at GMT+3 have this problem
and nix87 at GMT+1 - between us does not.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/scakemyer/plugin.video.quasar/issues/807#issuecomment-288354456>,
or mute the thread

**Comment by scakemyer** [comment]
I was trying to remember why I had to remove the air time checks, and it's because TMDB doesn't support air times for shows/episodes... That being said, we could do better timezone shifting for the dates themselves, but even if we did have air times, it wouldn't mean episodes would reliably have their description and screenshots updated right there and then; nothing prevents TMDB contributors to update the descriptions *days* before it airs either (screenshots are another matter obviously).

Still, I think we should be fine querying Trakt for the air times, since we wouldn't have to do so for every show when listing them, and would only do it once when going to a show's seasons/episodes. Only thing that would get slowed down doing this is library syncing, but since it's done in the background, I think that would be an acceptable compromise.

**Comment by KillerJoeBR** [comment]
Don't know if this relevant but just now an episode of modern family was added like a full week before air time.
